packages/client/src/workqueue/reducer.ts

```diff
@@ -129,6 +133,84 @@ export function updateRegistrarWorkqueue(
   }
 }
 
+async function getFilteredDeclarations(
+  workqueue: IWorkqueue,
+  getState: () => IStoreState
+): Promise<{
+  currentlyDownloadedDeclarations: IDeclaration[]
+  unassignedDeclarations: IDeclaration[]
+}> {
+  const state = getState()
+  const scope = getScope(state)
+  const savedDeclarations = state.declarationsState.declarations
+
+  const workqueueDeclarations = Object.entries(workqueue.data).flatMap(
+    (queryData) => {
+      return queryData[1].results
+    }
+  ) as Array<GQLEventSearchSet | null>
+
+  let [currentlyDownloadedDeclarations, unassignedDeclarations]: [
+    IDeclaration[],
+    IDeclaration[]
+  ] = [[], []]
+
+  // for field agent, no declarations should be unassigned
+  // for registration agent, sent for approval declarations should not be unassigned
+
+  // for other agents, check if the status of workqueue declaration
+  // has changed and if that declaration is saved in the store
+  // also declaration should not show as unassigned when it is being submitted
+  if (scope?.includes('declare'))
+    return {
+      currentlyDownloadedDeclarations: savedDeclarations,
+      unassignedDeclarations
+    }
+
+  unassignedDeclarations = workqueueDeclarations
+    .filter(
+      (dec) =>
+        dec &&
+        hasStatusChanged(dec, savedDeclarations) &&
+        isNotSubmitting(dec, savedDeclarations)
+    )
+    .map((dec) => savedDeclarations.find((d) => d.id === dec?.id))
+    .filter((maybeDeclaration): maybeDeclaration is IDeclaration =>
+      Boolean(maybeDeclaration)
+    )
+
+  currentlyDownloadedDeclarations = savedDeclarations.filter(
+    (dec) =>
+      !unassignedDeclarations
+        .map((unassigned) => unassigned.id)
+        .includes(dec.id)
+  )
+
+  if (unassignedDeclarations.length > 0)
+    return {
+      currentlyDownloadedDeclarations: savedDeclarations,
+      unassignedDeclarations
+    }
```

Comment:
Could you explain this part please?

Reply:
if there are no unassigned declarations we don't have to update the indexDB, we can just return it. It actually should be, `(unassignedDeclarations.length === 0)`

Reply:
like we do for field agents previously